core/src/sumeragi/mod.rs

```diff
@@ -228,20 +216,14 @@ impl Sumeragi {
 
     /// Access node metrics.
     #[allow(clippy::expect_used)]
-    pub fn metrics_mutex_access(&self) -> std::sync::MutexGuard<Metrics> {
-        self.metrics_mutex
-            .lock()
-            .expect("`Mutex` in `metrics_mutex_access` poisoned. This should not happen, given that panics should stop Iroha.")
+    pub fn metrics_mutex_access(&self) -> MutexGuard<Metrics> {
```

Comment:
Normally you want to return something which `impl Deref<Metrics>` so that you're not confined to a single implementation of a Guard. For example, once it's no longer a `Mutex` guard,

Reply:
Another option is to return a short-lived reference. Then you don't need to leak the `MutexGuard` into the public API of another crate.